### ISSUE TITLE
symmetry

### ISSUE THREAD

**Issue by zhaishengfu** [tscohen/gconv_experiments#1]
![symmetry](https://user-images.githubusercontent.com/7630101/37825018-7917d33c-2ec9-11e8-8aab-e6002c701fa9.png)
Hello, I read your article, but i do not understant your definition of 'symmetrty', in my understanding, flip leaves the image invariant, right?? but this is not right for common images.

**Comment by tscohen** [comment]
Yes, this is a bit subtle. By "sampling grid" I mean the set of 2D points (i, j) where i and j are integers.  This set is called Z^2. This is not the same as the image itself, which is considered a function on the sampling grid, that associates with each point (i, j) a pixel value v = f(i, j). Thus, a symmetry of the sampling grid is simply a function that maps Z^2 onto Z^2 in an invertible manner. In practice, we further restrict this to isometries, i.e. distance preserving transformations, like shifts, rotations, and flips. Flipping corresponds to (i, j) -> (-i, j), which is an invertible map from Z^2 to Z^2.

We are only interested in symmetries of the sampling grid, because these allow us to implement the group convolution in a simple manner, by permuting the pixels of a filter in some special way. A group convolution for continuous rotations would be a bit more difficult to implement (but work is being done in this area).

Besides being a symmetry of the sampling grid, we want to consider only symmetries of the label function. This is the function that maps each image x to its true label L(x). A symmetry of the label function is a mapping g that satisfies L(gx) = L(x) for all images x. Usually, the label does not change if you shift, rotate, or flip the image. So these transformations are symmetries of the label function, and symmetries of the sampling grid.

Let me know if you have further questions.
